cypress-browser-agent-otel/cypress/e2e/spec.cy.js

```diff
@@ -0,0 +1,84 @@
+const SLService = require("../../services/sealightsService");
+const jwtDecode = require("jwt-decode").default;
+
+let testSession;
+let testStartTime;
+
+describe("Calculator app tests", async () => {
+  before(() => {
+    cy.task("readSealightsConfig").then(
+      async ({ buildSessionId, apiToken }) => {
+        const decoded = jwtDecode(apiToken); // Agent Token
+        const baseUrl = decoded["x-sl-server"]; // Base url of the backend
+        SLService.setConfig(baseUrl, apiToken, buildSessionId);
+
+        // Start a test session
+        const { testSessionId } = (await SLService.createTestSession()).data;
+        testSession = testSessionId;
+
+        cy.visit("http://localhost:3333");
+      }
+    );
+  });
+
+  beforeEach(() => {
+    cy.window().then((win) => {
+      // Set the correct baggage before a scenario runs with testName and the current testSessionId
+      const testName = Cypress.currentTest.title;
+      const customEvent = new CustomEvent("set:baggage", {
+        detail: {
+          "x-sl-test-name": testName,
+          "x-sl-test-session-id": testSession,
+        },
+      });
+      win.dispatchEvent(customEvent);
+      testStartTime = Date.now();
+    });
+  });
+
+  afterEach(async function () {
+    const testState = this.currentTest.state;
+    const testName = this.currentTest.title;
+    // Send test event to Sealights
+    await SLService.sendTestEvent(
+      testSession,
+      testName,
+      testStartTime,
+      Date.now(),
+      testState
+    );
+    testStartTime = undefined;
+
+    cy.window().then((win) => {
+      // Unset baggage after scenario
+      const customEvent = new CustomEvent("delete:baggage");
+      win.dispatchEvent(customEvent);
+    });
+  });
+
+  after((done) => {
+    // End the current test session after the running suite using Sealights Public API
+    SLService.endTestSession(testSession).finally(() => {
+      cy.window().then(async (win) => {
+        win.$SealightsAgent.sendAllFootprints().finally(done);
```

Comment:
I'm very curious how this code executes. My understanding is that everything inside the (win)=>{...} code happens inside the browser, but it calls done() which is defined on line 59 and runs inside the cypress context. 

Did you verify this is actually executing? For example, if sendAllFootprints() would now sleep for 10 seconds (just for testing), will this cause the test to wait as well?

Reply:
It's supposed to work like that but I will test this right now before confirming if it actually waits that long or might have timeout

Reply:
Verified with following code for future references:

```
after(function (done) {
  this.timeout(20000);
  // End the current test session after the running suite using Sealights Public API
  SLService.endTestSession(testSession).finally(() => {
    cy.window().then(async (win) => {
      win.$SealightsAgent.sendAllFootprints = () => {
        return new Promise((resolve) => {
          setTimeout(() => {
            resolve();
          }, 10000);
        });
      };
      win.$SealightsAgent.sendAllFootprints().finally(done);
    });
  });
});
```